source/common/common/BUILD

```diff
@@ -262,11 +279,21 @@ envoy_cc_library(
 envoy_cc_library(
     name = "version_lib",
     srcs = ["version.cc"],
+    hdrs = select({
+        "//bazel:manual_stamp": [":generate_version_linkstamp"],
+        # By default the header file is empty.
+        # This is done so that the definitions linked via the linkstamp rule don't cause collisions.
```

Comment:
Note that nothing prevents someone from running a rule that respects `linkstamp` _and_ uses the manual_stamp define. I'll document this in the docs if this is the approach we congeal on.